### PR TITLE
CredentialManager.obtain() helper for a standard credential workflow

### PR DESCRIPTION
**Pull request by mih**
See docstring for details.

This replaces a more specific implementation in the context of `create_sibling_webdav()`, and is now available as a more generic helper for use in other scenarios too.

### CODE REVIEW

**Comment by mih** [comment]
Windows is back! The only failure is caused by a glitch with `httpbin`

**Comment by mih** [comment]
Now complete with tests.

**Review by christian-monch — DISMISSED** [review]
LGTM

File: datalad_next/utils/credman.py [hunk 1]
```diff
@@ -553,6 +558,116 @@ def get_sort_key(x):
...
+        ------
+        ValueError
+          Raised when no matching credential could be found and none was
+          entered. Also raised, when a credential selected from a query result
+          or a manually entered one is missing any of the properties with a
+          name given in ``expected_props``.
+        """
+        cred = None
+        if not name:
+            if query_props:
```
Comment (adswa) [inline datalad_next/utils/credman.py:635]: Sorry if I'm missing something, but the docstring (first sentence) sounds as if I should be able to obtain a credential solely by prompt, without the need for query parameters. This conditional would prevent this, if I see it correctly

Comment (mih) [reply datalad_next/utils/credman.py:635]: Yes, that observation is correct. The type hint is needed such that the credential manager knows what to prompt for (some credentials need more than just a secret). This limitation could be lifted (to be able to just ask for a secret). Or we add docs, as you suggested.

I am always hesitant re examples. Unless executable (which most of the time required added/undesired complexity) they become invalid quickly, which often goes unnoticed. But I do see the issue.

Comment (mih) [reply datalad_next/utils/credman.py:656]: I have pushed some examples. Would that work for you?

Comment (adswa) [reply datalad_next/utils/credman.py:656]: Thx

**Pull request merged**